### PR TITLE
Created the fibonacci function

### PR DESCRIPTION
**Pull request by gagan-maheshwari**
## Description
Wrote the fibonacci function which generates fibonacci numbers upto n numbers.

### Screenshots / Videos

https://user-images.githubusercontent.com/84700028/127369745-c7b0a078-7b78-4b9a-8be9-f81fc6e067c6.mov

## Database schema changes
None

### Automated test cases added
- None

### Manual test cases run

**Test Case 1**
1. Call the printFibonacciNumbers with a positive number, eg. 5
2. Run: `node index.js`
3. Verify ER1.

Expected Result:
1. The output should be the following:
> Fibonacci Series:
> 0
> 1
> 1
> 2
> 3
---
**Test Case 2**
1. Call the printFibonacciNumbers with a zero or a negative number, eg. 0 or -5
2. Run: `node index.js`
3. Verify ER1.

Expected Result:
1. The output should be the following:
> Number cannot be less than 1
---
**Test Case 3**
1. Call the printFibonacciNumbers with a floating point number, eg. 3.8
2. Run: `node index.js`
3. Verify ER1.

Expected Result:
1. The output should be the following:
> Only integers are permitted! Please try again.
---
**Test Case 4**
1. Call the printFibonacciNumbers with a string, eg. "asdf"
2. Run: `node index.js`
3. Verify ER1.

Expected Result:
1. The output should be the following:
> Only integers are permitted! Please try again.
---

## Limitations
1. With an input like `010` which seems like the number `10`, the output will go awry. Reason being that the input `010` is interpreted as an octal number.
2. On similar grounds as the above point, an input like `0x20`, being a hexadecimal number will evaluate to `32` (in decimal number system). Hence it would print 32 numbers in the series.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    {
+        console.log(n1);
+
+        nextTerm = n1 + n2;
+        n1 = n2;
+        n2 = nextTerm;
+    }
+}
+
+printFibonacciNumbers(5);
```
Comment (jjalan) [inline index.js:29]: add end of file (new line).

File: index.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    let n1 = 0, n2 = 1, nextTerm;
+    
+    for(let i = 0; i < n; i++)
+    {
+        console.log(n1);
+
+        nextTerm = n1 + n2;
+        n1 = n2;
+        n2 = nextTerm;
+    }
```
Comment (jjalan) [inline index.js:26]: use 2 instead of 4 spaces

File: index.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
+const printFibonacciNumbers = (n) =>
+{
+    if(!Number.isInteger(n))
```
Comment (jjalan) [inline index.js:3]: We want to use `if(!Number.isInteger(n)) {` format to make code succint. `{` on new line generate quite verbose code

**Review by jjalan — APPROVED** [review]
Ok to merge after addressing formatting issue.

File: index.js (hunk 1 above)
Comment (gagan-maheshwari) [reply index.js]: Done

File: index.js (hunk 2 above)
Comment (gagan-maheshwari) [reply index.js]: Done

File: index.js (hunk 3 above)
Comment (gagan-maheshwari) [reply index.js]: Done

**Pull request merged**